### PR TITLE
Add tool with filepath input and flow example with this tool to used for the filepath doc

### PR DESCRIPTION
**Pull request by liuyuhang13**
Add a tool with file path input;
Add a flow using this tool;

All these material will be used to compose the 'Using File Path in Tool Input' doc. Refer this PR: https://github.com/microsoft/promptflow/pull/763

### CODE REVIEW

File: docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Use FilePath as input in python tool and custom tool
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md:1]: Use file path as tool input

File: docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+1. Download private version promptflow SDK
+    ```
+    pip install "promptflow==0.0.106113823" --extra-index-url https://azuremlsdktestpypi.azureedge.net/test-promptflow/
+    ```
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX: 
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath in python tool
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md:17]: The gen meta difference between package tool (custom tool you mentioned) and python tool is not what we focus in this doc. From tool interface perspetive, they are the same. Our focus of this doc is how to write a tool with file path type.

I think you could add a tool with file path input in hello-world/tools rather than modify existing one.

File: docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   2. Use FilePath as Input
+   3. Consume Input
+
+   ![update custom tool](../../media/how-to-guides/develop-a-tool/update_custom_tool_with_file_path.png)
+3. Generate tool metadata, build and install tool package in VSCode Extension
+4. Go to the extension and open one flow folder. Click 'flow.dag.yaml' and preview the flow. Next, click `+ More` button and select your tool `Hello World Tool` 
+   ![use custom tool](../../media/how-to-guides/develop-a-tool/use_custom_tool.png)
+5. Click pencil after file_path, select `existing file` or `new file` as input, here select new file 
+   ![file path use new file](../../media/how-to-guides/develop-a-tool/file_path_new_file.png)
+   input_file.txt as input:
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md:50]: For example, use `.py` as input rather than `.txt`.

**Review by chjinche — COMMENTED** [review]
may consider adding such flow to `examples` folder for practical use case powered by file_path.
https://msdata.visualstudio.com/Vienna/_git/PromptFlow?version=GBchjinche/add_sk_example&path=/src/promptflow-sdk/tests/test_configs/flows/semantic_kernel_planner

File: docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md (hunk 1 above)
Comment (liuyuhang13) [reply docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md]: updated.

File: docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md (hunk 2 above)
Comment (liuyuhang13) [reply docs/how-to-guides/develop-a-tool/use-file-path-in-python-tool-and-custom-tool.md]: updated

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    pip install "promptflow==0.0.106113823" --extra-index-url https://azuremlsdktestpypi.azureedge.net/test-promptflow/
+    ```
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX:
+
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath as tool input in 'hello-world' tool
+
+For details of how to create and use custom tool, please refer to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:21]: Where is how-to-create-and-use-your-own-tool-package.md ?

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    python <path-to-scripts>\tool\generate_tool_package_template.py --case file_path --destination hello-world-proj --package-name hello-world --tool-name hello_world_tool --function-name get_sqrt    
+    ```
+
+2. Go to hello-world-proj folder, check your tool code.
+
+   The demo use a sematic_kernel case, refer to [run native code with Semantic Kernel](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/native-functions/using-the-skfunction-decorator?tabs=python).
+
+   ![update custom tool](../../media/how-to-guides/develop-a-tool/generate_custom_tool_with_file_path.png)
+
+3. Build and install tool package in VSCode Extension, according to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:35]: how-to-create-and-use-your-own-tool-package.md is not exist

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Use FilePath as tool input
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:1]: Better to introduce the What is FilePath type and its capability with a screenshot to let user know that it can pick up a file in UI.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+   ```
+
+6. Click pencil after file_path, select `existing file` or `new file` as input, here select exsiting file demo_input.py from step 4 as input:
+
+   ![file path use existing file](../../media/how-to-guides/develop-a-tool/select_input_file.png)
+
+7. Run your flow, the demo calculate sqrt of input_str using skill in step 4.
+   ![flow output](../../media/how-to-guides/develop-a-tool/custom_tool_output.png)
+
+## FAQ
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:89]: Does it only support the file in the flow folder ？ Or the file can be placed in anywhere ?

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+Since FilePath is still in test currently, you need to install private version SDK and Extension.
+
+1. Download private version promptflow SDK
+
+    ```bash
+    pip install "promptflow==0.0.106113823" --extra-index-url https://azuremlsdktestpypi.azureedge.net/test-promptflow/
+    ```
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX:
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:15]: For install promptflow extension, better to keep consistent with create-and-use-tool-package.md:
image 
and use this link : https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow since it is more official.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath as tool input in 'hello-world' tool
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:19]: ## Use FilePath as tool input ? Remove the "in hello-world tool'

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+For details of how to create and use custom tool, please refer to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
+
+1. Create a tool project hello-world-proj with `--case file_input` to generate demo case of file_path as tool input.
+
+    ```bash
+    python <path-to-scripts>\tool\generate_tool_package_template.py --case file_path --destination hello-world-proj --package-name hello-world --tool-name hello_world_tool --function-name get_sqrt    
+    ```
+
+2. Go to hello-world-proj folder, check your tool code.
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:29]: Go to hello-world-proj folder, it uses FilePath as tool input in hello_world_tool.py

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 4 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:21]: Also mentioned that "Here we take the hello-world-proj' as an example. You can create your own tool package by following [Create and Use Tool Package](create-and-use-tool-package.md)"

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 5 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:35]: Step 3 and the rest can be another section: Use tool with FilePath input in VSCode Extension

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 6 above)
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:1]: Will prefer to use `file path`

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX:
+
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath as tool input in 'hello-world' tool
+
+For details of how to create and use custom tool, please refer to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
+
+1. Create a tool project hello-world-proj with `--case file_input` to generate demo case of file_path as tool input.
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:23]: 1-4 are all steps of tool package creation, which can be replaced simply by a reference link to create tool package doc.
Would suggest add an example tool to https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart/my_tool_package/tools, and start from the tool definition and show what tool author should do.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+   ![use hello world tool](../../media/how-to-guides/develop-a-tool/select_hello_world_tool.png)
+
+5. Prepare your own python script, `demo_input.py` as file_input.
+
+   Here the input demos a custom sqrt func, for more details, refer to [Use the SKFunction decorator to define a native function](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/native-functions/using-the-skfunction-decorator?tabs=python#use-the-skfunction-decorator-to-define-a-native-function)
+
+   ```python
+    import math
+    
+    from semantic_kernel.skill_definition import sk_function, sk_function_context_parameter
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:61]: this example tool requires `semantic-kernel` package, where do you add the dependency? Maybe we can add a simple example.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+   ```
+
+4. Go to the extension and create new flow. Click 'empty flow' and preview the flow. Next, click `+ More` button and select your tool `Hello World Tool`
+
+   ![create empty flow](../../media/how-to-guides/develop-a-tool/create_empty_flow.png)
+   ![use hello world tool](../../media/how-to-guides/develop-a-tool/select_hello_world_tool.png)
+
+5. Prepare your own python script, `demo_input.py` as file_input.
+
+   Here the input demos a custom sqrt func, for more details, refer to [Use the SKFunction decorator to define a native function](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/native-functions/using-the-skfunction-decorator?tabs=python#use-the-skfunction-decorator-to-define-a-native-function)
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:56]: Need to share more context/background of the sample you made.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+    ```
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX:
+
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath as tool input in 'hello-world' tool
+
+For details of how to create and use custom tool, please refer to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
+
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:22]: We need code sample to teach user how to write a tool with file path type input.

Comment (chjinche) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:22]: Prefer to paste code rather than screenshot.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 6 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:1]: change it to Use file path as tool input

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 11 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md]: Add a tool to it.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 13 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md]: change the sample to an easy one.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md (hunk 14 above)
Comment (dorisjoy) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md]: paste code here

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
+# Use file path as tool input
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:1]: Using File Path as a Tool Input

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+User sometimes need to reference their local files within a tool to implement some specific logic. To simplify this, we've introduced the FilePath input type. This input type enables users to either select an existing file or create a new one, then pass it to a tool, enabling the tool to access the file's content.
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:3]: subtle edit:
Users sometimes need to reference local files within a tool to implement specific logic. To simplify this, we've introduced the `FilePath` input type. This input type enables users to either select an existing file or create a new one, then pass it to a tool, allowing the tool to access the file's content.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+In this guide, we will provide a detailed walkthrough on how to use FilePath as a tool input. We will also demostrate the user experience when utilizing this type of tool within a flow.
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:5]: `FilePath`

Comment (chjinche) [reply docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:5]: Let's wrap other places using FilePath as `FilePath`.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+    ```bash
+    pip install "promptflow==0.1.0b8.dev2" --extra-index-url https://azuremlsdktestpypi.azureedge.net/promptflow
+    ```
+
+## Use file path as tool input
+
+Here we use [an existing tool package](../../../examples/tools/tool-package-quickstart/my_tool_package) as an example. If you want to create your owner tool, please refer to [create and use tool package](https://github.com/microsoft/promptflow/blob/main/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#create-custom-tool-package).
+
+1. Add FilePath input for your tool, refer to [this tool](../../../examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_file_path_input.py).
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:19]: 1. Add a `FilePath` input for your tool, like in [this example](../../../examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_file_path_input.py):

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+    # 2. add a FilePath input for your tool method
+    @tool
+    def my_tool(input_file: FilePath, input_text: str) -> str:
+        # 3. customise your own code to handle and use the input_file here
+        new_module = importlib.import_module(Path(input_file).stem)
+    
+        return new_module.hello(input_text)   
+    ```
+
+2. FilePath input format in a tool yaml, refer to [this tool yaml](../../../examples/tools/tool-package-quickstart/my_tool_package/yamls/tool_with_file_path_input.yaml)
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:37]: 2. `FilePath` input format in a tool YAML, like in [this example](../../../examples/tools/tool-package-quickstart/my_tool_package/yamls/tool_with_file_path_input.yaml):

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+   ```yaml
+    my_tool_package.tools.my_tool_1.tool_with_file_path_input:
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:40]: my_tool_package.tools.tool_with_file_path_input:?

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+    ```
+
+    > [!Note] tool yaml file can be generated using a python script. For further details, please refer to [create custom tool package](https://github.com/microsoft/promptflow/blob/main/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#create-custom-tool-package).
+
+
+## Use tool with a file path input in VS Code extension
+Follow steps to [build and install your tool package](https://github.com/jiazengcindy/mpromptflow/blob/jiazeng/modify_icon_doc/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#build-and-share-the-tool-package) and [use your tool from VS Code extension](https://github.com/jiazengcindy/mpromptflow/blob/jiazeng/modify_icon_doc/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#use-your-tool-from-vscode-extension).
+
+Here we use an existing flow to demonstrate the experience, open [this flow]((../../../examples/flows/standard/flow-use-tool-with-file-path-input/flow.dag.yaml)) in VS Code extension:
+- There is a node named "Tool_with_FilePath_Input" has a file_path type input: input_file;
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:63]: - There is a node named "Tool_with_FilePath_Input" with a `file_path` type input called `input_file`.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+- Click the picker icon to trigger the UI to select an existing file or create a new file and use it as input;
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:64]: - Click the picker icon to open the UI for selecting an existing file or creating a new file to use as input.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+
+   ![](../../media/how-to-guides/develop-a-tool/use_file_path_in_flow.png)
+
+## FAQ
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:68]: Here is an improved FAQ section:

## FAQs

### What are some practical use cases for this feature?

The `FilePath` input enables several useful workflows:

1. **Dynamically load modules** - As shown in the demo, you can load a Python module from a specific script file selected by the user. This allows flexible custom logic. 

2. **Load arbitrary data files** - The tool can load data from files like .csv, .txt, .json, etc. This provides an easy way to inject external data into a tool.

So in summary, `FilePath` input gives tools flexible access to external files provided by users at runtime. This unlocks many useful scenarios like the ones above.

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+      type: python   
+    ```
+
+    > [!Note] tool yaml file can be generated using a python script. For further details, please refer to [create custom tool package](https://github.com/microsoft/promptflow/blob/main/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#create-custom-tool-package).
+
+
+## Use tool with a file path input in VS Code extension
+Follow steps to [build and install your tool package](https://github.com/jiazengcindy/mpromptflow/blob/jiazeng/modify_icon_doc/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#build-and-share-the-tool-package) and [use your tool from VS Code extension](https://github.com/jiazengcindy/mpromptflow/blob/jiazeng/modify_icon_doc/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#use-your-tool-from-vscode-extension).
+
+Here we use an existing flow to demonstrate the experience, open [this flow]((../../../examples/flows/standard/flow-use-tool-with-file-path-input/flow.dag.yaml)) in VS Code extension:
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:62]: Let's name the sample flow as `filepath-input-tool-showcase`

File: examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_file_path_input.py [hunk 25]
```diff
@@ -0,0 +1,12 @@
...
+from promptflow import tool
+from promptflow.contracts.types import FilePath
+
+
+@tool
+def my_tool(input_file: FilePath, input_text: str) -> str:
+    # customise your own code to handle and use the input_file here
+    new_module = importlib.import_module(Path(input_file).stem)
+
+    return new_module.hello(input_text)
```
Comment (chjinche) [inline examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_file_path_input.py:12]: In flow.dag you use `convert_image_to_data_url.py` as example, but this py does not have `hello` function?

File: examples/flows/standard/flow-use-tool-with-file-path-input/flow.dag.yaml [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+    reference: ${Tool_with_FilePath_Input.output}
+nodes:
+- name: Tool_with_FilePath_Input
+  type: python
+  source:
+    type: package
+    tool: my_tool_package.tools.my_tool_1.tool_with_file_path_input
+  inputs:
+    input_text: ${inputs.input}
+    input_file: ..\..\..\..\scripts\tool\convert_image_to_data_url.py
```
Comment (chjinche) [inline examples/flows/standard/flow-use-tool-with-file-path-input/flow.dag.yaml:17]: should be the file in the flow? Maybe improve the name `file_as_file_path_input.py`.

File: examples/flows/standard/flow-use-tool-with-file-path-input/file_as_file_path_input.py [hunk 27]
```diff
@@ -0,0 +1,3 @@
+def hello(input_text: str) -> str:
```
Comment (chjinche) [inline examples/flows/standard/flow-use-tool-with-file-path-input/file_as_file_path_input.py:1]: `file_path_input_source.py`?

File: docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md [hunk 28]
```diff
@@ -0,0 +1,76 @@
+# Using File Path as tool input
```
Comment (chjinche) [inline docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md:1]: "Tool Input"? Just keep consistent with other doc title.

File: examples/flows/standard/filepath-input-tool-showcase/flow.dag.yaml [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+outputs:
+  output:
+    type: string
+    reference: ${Tool_with_FilePath_Input.output}
+nodes:
+- name: Tool_with_FilePath_Input
+  type: python
+  source:
+    type: package
+    tool: my_tool_package.tools.my_tool_1.tool_with_file_path_input
```
Comment (chjinche) [inline examples/flows/standard/filepath-input-tool-showcase/flow.dag.yaml:14]: my_tool_package.tools..tool_with_file_path_input

File: examples/flows/standard/flow-use-tool-with-file-path-input/file_as_file_path_input.py (hunk 27 above)
Comment (dorisjoy) [reply examples/flows/standard/flow-use-tool-with-file-path-input/file_as_file_path_input.py]: change it to hello_method.py

**Comment by crazygao** [comment]
Please add a readme following examples/contributing.md

**Review by brynn-code — APPROVED** [review]

**Review by crazygao — APPROVED** [review]
